src/python/pants/backend/python/dependency_inference/rules_test.py

```diff
@@ -499,6 +499,81 @@ def mk_imports_rule_runner(more_rules: Iterable) -> RuleRunner:
     )
 
 
+def test_infer_python_ignore_unowned_imports(imports_rule_runner: RuleRunner, caplog) -> None:
+    """Test handling unowned imports that are set explicitly to be ignored."""
+    imports_rule_runner.write_files(
+        {
+            "src/python/cheesey.py": dedent(
+                """\
+                    import unknown_python_requirement
+                    import project.application.generated
+                    import project.application.generated.loader
+                    from project.application.generated import starter
+                    import project.application.develop.client
+                    import project.application.development
+                """
+            ),
+            "src/python/BUILD": "python_sources()",
+        }
+    )
+
+    def run_dep_inference(
+        unowned_dependency_behavior: str, ignored_paths: tuple[str, ...] = tuple()
+    ) -> InferredDependencies:
+        imports_rule_runner.set_options(
+            [
+                f"--python-infer-unowned-dependency-behavior={unowned_dependency_behavior}",
+                f"--python-infer-ignore-unowned-imports={str(list(ignored_paths))}",
+            ],
+            env_inherit=PYTHON_BOOTSTRAP_ENV,
+        )
+        target = imports_rule_runner.get_target(
+            Address("src/python", relative_file_path="cheesey.py")
+        )
+        return imports_rule_runner.request(
+            InferredDependencies,
+            [
+                InferPythonImportDependencies(
+                    PythonImportDependenciesInferenceFieldSet.create(target)
+                )
+            ],
+        )
+
+    run_dep_inference("warning")
+    assert len(caplog.records) == 1
+    assert (
```

Comment:
is it worth extracting a helper for asserting when owners are not found? There are only 2 uses, but it would be clearer to see something like `assert_owners_not_found_error(target="src/python/cheesey.py", not_found=[...])`

Reply:
Sure thing, great idea, done :)